api/types/swarm/task.go

```diff
@@ -71,8 +72,30 @@ type TaskSpec struct {
 
 // Resources represents resources (CPU/Memory).
 type Resources struct {
-	NanoCPUs    int64 `json:",omitempty"`
-	MemoryBytes int64 `json:",omitempty"`
+	NanoCPUs         int64              `json:",omitempty"`
+	MemoryBytes      int64              `json:",omitempty"`
+	GenericResources []*GenericResource `json:",omitempty"`
+}
+
+// GenericResource represents a "user defined" resource which can
+// be either an integer (e.g: SSD=3) or a string (e.g: SSD=sda1)
+type GenericResource struct {
+	Str      *GenericString   `json:",omitempty"`
```

Comment:
Yep Str means string.
A resource can either be described by an int (SSD=3) or through a String (GPU=UUID-1)

This is described in more details in the design doc: https://github.com/docker/swarmkit/blob/master/design/generic_resources.md#use-cases